--- a/watch/watch.py      2018-11-06 20:41:11.000000000 +0000
+++ b/watch/watch.py      2018-12-10 11:56:47.558309526 +0000
@@ -81,15 +81,18 @@
         js['raw_object'] = js['object']
         if return_type:
             obj = SimpleNamespace(data=json.dumps(js['raw_object']))
+            version = None
             js['object'] = self._api_client.deserialize(obj, return_type)
             if hasattr(js['object'], 'metadata'):
-                self.resource_version = js['object'].metadata.resource_version
+                version = js['object'].metadata.resource_version
             # For custom objects that we don't have model defined, json
             # deserialization results in dictionary
             elif (isinstance(js['object'], dict) and 'metadata' in js['object']
                   and 'resourceVersion' in js['object']['metadata']):
-                self.resource_version = js['object']['metadata'][
+                version = js['object']['metadata'][
                     'resourceVersion']
+            if version is not None:
+                self.resource_version = version
         return js
 
     def stream(self, func, *args, **kwargs):
