--- a/watch/watch.py      2018-12-10 14:37:19.999580768 +0000
+++ b/watch/watch.py      2018-12-10 14:38:36.104930476 +0000
@@ -83,13 +83,21 @@
             obj = SimpleNamespace(data=json.dumps(js['raw_object']))
             js['object'] = self._api_client.deserialize(obj, return_type)
             if hasattr(js['object'], 'metadata'):
-                self.resource_version = js['object'].metadata.resource_version
+                version = js['object'].metadata.resource_version
             # For custom objects that we don't have model defined, json
             # deserialization results in dictionary
             elif (isinstance(js['object'], dict) and 'metadata' in js['object']
                   and 'resourceVersion' in js['object']['metadata']):
-                self.resource_version = js['object']['metadata'][
+                version = js['object']['metadata'][
                     'resourceVersion']
+        elif (isinstance(js['object'], dict) and 'metadata' in js['object']
+              and 'resourceVersion' in js['object']['metadata']):
+            version = js['object']['metadata'][
+                'resourceVersion']
+            
+        if version is not None and version.lower() != 'none':
+            self.resource_version = version
+
         return js
 
     def stream(self, func, *args, **kwargs):
